### PR TITLE
Feature/tcvp 1701

### PR DESCRIPTION
**Pull request by lfdgov**
# Description

This PR includes the following proposed change(s):

- JIRA ticket 1701 and subtasks
- necessary changes to help convert to keycloak gold
- formatting improvements for non-existent counts

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

- Converted several written reasons disputes to requires court hearing
- tested assign / unassign of JJ Disputes and filtering

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [x] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/frontend/staff-portal/src/app/components/jj-dispute-info/jj-count/jj-count.component.ts [hunk 1]
```diff
@@ -71,22 +72,22 @@ export class JJCountComponent implements OnInit {
...
+          probationConditions:  [{ value: null, disabled: !this.jjDisputedCount }],
+          drivingProhibition:  [{ value: null, disabled: !this.jjDisputedCount }],
+          drivingProhibitionMVASection:  [{ value: null, disabled: !this.jjDisputedCount }],
+          dismissed:  [{ value: null, disabled: !this.jjDisputedCount }],
+          forWantOfProsecution:  [{ value: null, disabled: !this.jjDisputedCount }],
+          withdrawn:  [{ value: null, disabled: !this.jjDisputedCount }],
+          abatement:  [{ value: null, disabled: !this.jjDisputedCount }],
+          stayOfProceedingsBy:  [{ value: null, disabled: !this.jjDisputedCount }],
+          other:  [{ value: null, disabled: !this.jjDisputedCount }],
         })
```
[the remark this answers is not in the document]
Comment (lfdgov) [reply src/frontend/staff-portal/src/app/components/jj-dispute-info/jj-count/jj-count.component.ts:91]: line 111 to disable the form was not having the same effect -- individual form controls were not being disabled, which is why I added this code.

File: src/frontend/staff-portal/src/app/components/jj-dispute-info/jj-dispute/jj-dispute.component.ts [hunk 2]
```diff
@@ -75,12 +78,41 @@ export class JJDisputeComponent implements OnInit {
     });;
   }
 
-  onSubmit(): void {
-    this.lastUpdatedJJDispute.status = JJDisputeStatus.Confirmed;  // Send to VTC Staff for review
+  public onSubmit(): void {
```
[the remark this answers is not in the document]
Comment (lfdgov) [reply src/frontend/staff-portal/src/app/components/jj-dispute-info/jj-dispute/jj-dispute.component.ts:81]: Hmm I didnt add the word public here -- must be from merging with something.

File: src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/controller/v1_0/JJDisputeController.java [hunk 3]
```diff
@@ -173,6 +173,31 @@ public ResponseEntity<JJDispute> reviewJJDispute(@PathVariable String ticketNumb
...
+		@ApiResponse(responseCode = "400", description = "Bad Request."),
+		@ApiResponse(responseCode = "404", description = "JJDispute record not found. Update failed."),
+		@ApiResponse(responseCode = "405", description = "A JJDispute status can only be set to REVIEW iff status is NEW or VALIDATED and the remark must be <= 256 characters. Update failed."),
+		@ApiResponse(responseCode = "500", description = "Internal server error occured.")
+	})
+	@PutMapping("/dispute/{ticketNumber}/requirecourthearing")
+	public ResponseEntity<JJDispute> requireCourtHearingJJDispute(@PathVariable String ticketNumber,
+			@RequestBody @Size(max = 256) String remark,
+			Principal principal) {
+		logger.debug("PUT /dispute/{id}/requirecourthearing called");
```
Comment (KDNeufeld) [inline src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/controller/v1_0/JJDisputeController.java:196]:
```suggestion
		logger.debug("PUT /dispute/{}/requirecourthearing called", ticketNumber);
```

File: src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/ViolationTicketMapperTest.java [hunk 4]
```diff
@@ -15,7 +15,7 @@
 public class ViolationTicketMapperTest extends BaseTestSuite{
 
 	private ViolationTicketMapper violationTicketMapper;
-
+
```
Comment (KDNeufeld) [inline src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/ViolationTicketMapperTest.java:18]: If you update your IDE to always trim trailing whitespace on save, we won't be flipping back and forth with these whitespace changes.

Comment (lfdgov) [reply src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/ViolationTicketMapperTest.java:18]: My Eclipse IDE has this setting -- unknown why this one slipped through

File: src/frontend/staff-portal/src/app/components/jj-dispute-info/jj-count/jj-count.component.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Sharkww) [reply src/frontend/staff-portal/src/app/components/jj-dispute-info/jj-count/jj-count.component.ts:91]: just suggestion, you can keep it as is

**Pull request merged**